### PR TITLE
Update PatchResponseCodes rule name, implementation, tests

### PR DESCRIPTION
**Pull request by AkhilaIlla**
Update PatchResponseCode rule to match with the post/put/delete response codes implementation

### CODE REVIEW

File: docs/patch-response-codes.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+LRO PATCH must have 200 and 202 and default return codes. They also must not have other response codes.
+
+
+## Description
+
+Synchronous PATCH must have 200 return code and LRO PATCH must have 200 and 202 return codes.
+
+## How to fix the violation
+
+For an Synchronous PATCH add 200 and default return codes and make sure they don't have other response codes.
```
Comment (rkmanda) [inline docs/patch-response-codes.md:27]: >an [](http://example.com/codeflow?start=4&length=2)

a

File: docs/patch-response-codes.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+### Example
+
+The following would be a valid SYNC PATCH:
+
+```json
+...
+patch {
+  "responses": {
+    "200": {
+      "description": "Operation completed"
```
Comment (rkmanda) [inline docs/patch-response-codes.md:39]: >"description": "Operation completed" [](http://example.com/codeflow?start=6&length=36)

mention a schema here

File: docs/patch-response-codes.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+```json
+...
+patch{
+  "responses": {
+      "202": {
+        "description": "Operation accepted",
+      },
+      "200": {
+        "description": "Operation completed"
```
Comment (rkmanda) [inline docs/patch-response-codes.md:62]: >   "description": "Operation completed" [](http://example.com/codeflow?start=5&length=39)

mention a schema here

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: docs/patch-response-codes.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+ARM OpenAPI(swagger) specs
+
+## Related ARM Guideline Code
+
+- RPC-Patch-V1-06
+
+## Output Message
+
+Synchronous PATCH must have 200 and default return codes. They also must not have other response codes.
```
Comment (bdefoy) [inline docs/patch-response-codes.md:17]:
```suggestion
Synchronous PATCH operations must have `200` and `default` return codes. They also must not have other response codes.
```

File: docs/patch-response-codes.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+LRO PATCH must have 200 and 202 and default return codes. They also must not have other response codes.
```
Comment (bdefoy) [inline docs/patch-response-codes.md:18]:
```suggestion
LRO PATCH operations must have `200`, `202`, and `default` return codes. They also must not have other response codes.
```

File: docs/patch-response-codes.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+Synchronous PATCH must have 200 return code and LRO PATCH must have 200 and 202 return codes.
+
+## How to fix the violation
+
+For a Synchronous PATCH add 200 and default return codes and make sure they don't have other response codes.
+For LRO PATCH add 200, 202 and default return codes and make sure they don't have other response codes.
+
+### Example
+
+The following would be a valid SYNC PATCH:
```
Comment (bdefoy) [inline docs/patch-response-codes.md:32]: also add invalid examples

File: docs/patch-response-codes.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+LRO PATCH must have 200 and 202 and default return codes. They also must not have other response codes.
+
+
+## Description
+
+Synchronous PATCH must have 200 return code and LRO PATCH must have 200 and 202 return codes.
+
+## How to fix the violation
+
+For a Synchronous PATCH add 200 and default return codes and make sure they don't have other response codes.
```
Comment (rkmanda) [inline docs/patch-response-codes.md:27]: >Synchronous [](http://example.com/codeflow?start=6&length=11)

nit: low case 's'

File: docs/patch-response-codes.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+```json
+...
+patch{
+  "responses": {
+      "202": {
+        "description": "Operation accepted",
+        "schema": {
+          "$ref": "#/definitions/FooResource",
+        },
```
Comment (rkmanda) [inline docs/patch-response-codes.md:65]: 202 must not have a schema. pl remove

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**